File: Xtra-Kick(Alpha).apk_Decompiler.com/sources/com/github/andreyasadchy/xtra/ui/games/GamesViewModel.java
package com.github.andreyasadchy.xtra.ui.games;

import androidx.lifecycle.p1;
import androidx.lifecycle.w1;
import e.a;
import e6.x;
import hb.h0;
import javax.inject.Inject;
import kb.i;
import kb.m0;
import n3.r3;
import xa.j;

public final class GamesViewModel extends w1 {

    /* renamed from: d  reason: collision with root package name */
    public final x f3430d;

    /* renamed from: e  reason: collision with root package name */
    public final m0 f3431e = h0.s((i) new a(new r3(30, 10, 30), (wa.a) new p1(13, this)).f4852i, h0.I0(this));

    @Inject
    public GamesViewModel(x xVar) {
        j.f("apiRepository", xVar);
        this.f3430d = xVar;
    }
}

File: app/src/main/java/com/github/andreyasadchy/xtra/ui/games/GamesViewModel.kt
package com.github.andreyasadchy.xtra.ui.games

import android.content.Context
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import androidx.paging.Pager
import androidx.paging.PagingConfig
import androidx.paging.cachedIn
import com.github.andreyasadchy.xtra.model.ui.Tag
import com.github.andreyasadchy.xtra.repository.KickRepository
import com.github.andreyasadchy.xtra.repository.datasource.GamesDataSource
import com.github.andreyasadchy.xtra.util.C
import com.github.andreyasadchy.xtra.util.KickApiHelper
import com.github.andreyasadchy.xtra.util.prefs
import dagger.hilt.android.lifecycle.HiltViewModel
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.flatMapLatest
import javax.inject.Inject

@HiltViewModel
class GamesViewModel @Inject constructor(
    @ApplicationContext applicationContext: Context,
    private val kickRepository: KickRepository,
) : ViewModel() {

    val filter = MutableStateFlow<Filter?>(null)
    val filtersText = MutableStateFlow<CharSequence?>(null)

    val tags: Array<Tag>
        get() = filter.value?.tags ?: emptyArray()

    @OptIn(ExperimentalCoroutinesApi::class)
    val flow = filter.flatMapLatest { filter ->
        Pager(
            PagingConfig(pageSize = 30, prefetchDistance = 10, initialLoadSize = 30)
        ) {
            GamesDataSource(
                tags = tags.ifEmpty { null }?.mapNotNull { it.id },
                kickRepository = kickRepository,
                apiPref = listOf(C.KICK),
            )
        }.flow
    }.cachedIn(viewModelScope)

    fun setFilter(tags: Array<Tag>?) {
        filter.value = Filter(tags)
    }

    class Filter(
        val tags: Array<Tag>?,
    )
}
